clash-lib/src/Clash/Normalize/Transformations.hs

```diff
@@ -2472,6 +2482,65 @@ separateArguments ctx@(TransformContext is0 _) e@(Lam b eb0) = do
         isN1 = extendInScopeSet isN0 x'
     in  (isN1,x')
 
+  -- When splitting up a single argument into multiple arguments (see docs of
+  -- 'separateArguments') we should make sure to update TopEntity annotations
+  -- accordingly. See: https://github.com/clash-lang/clash-compiler/issues/1033
+  splitTopAnn
+    :: HasCallStack
+    => Int
+    -- ^ Number of ports to insert
+    -> (TopEntityT, PortName)
+    -- ^ TopEntity to insert ports in / port name of
+    -> NormalizeSession ()
+  splitTopAnn n (t0@(TopEntityT id_ (Just s@(Synthesize{t_inputs})) _tb), portName) =
+    case portName of
+      PortProduct nm portNames0 ->
+        let
+          (pre, tail -> post) = splitAt (length cctx) t_inputs
+          newPortNames = map (PortName . show) [(0::Int)..]
+          portNames1 = take n (map (prependName nm) (portNames0 ++ newPortNames))
+          t1 = t0{topAnnotation=Just s{t_inputs=pre ++ portNames1 ++ post}}
+        in do
+          (cf, _) <- Lens.use curFun
+          topEntities Lens.%= extendVarEnv cf t1
```

Comment:
This is the real reason I'm here :wink:

I know this kind of thing appears in other places, but I really think qualified operators make things less readable. We also (at least with lenses) are relatively inconsistent with whether we qualify them or not.

(Also by no means are these comments a review in any way, it's just me seeing some of my Haskell bugbears)

Reply:
Is this about `Lens.use`, `Lens.%=`, or both?

Reply:
Just `Lens.%=`. I think (especially for `lens` having such generic names) it's nice to qualify functions, but have operators in an explicit unqualified import list. Mostly because qualified operators are weird to read, and sometimes confusing (if the operator starts with a dot for instance)

Reply:
Plus we usually only use about 3 lens operators at most in a particular file, so mentioning them explicitly in an import list doesn't make things too horrible